### DISCUSSION TITLE
ScriptReference/Transform-localToWorldMatrix

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#507] [Notes]
# ScriptReference/Transform-localToWorldMatrix

https://docs.unity3d.com/ScriptReference/Transform-localToWorldMatrix.html

**Comment by HeliosDoubleSix** [comment]
To do the reverse as this is read only:

yourTransform.position = new Vector3( yourMatrix.m03, yourMatrix.m13, yourMatrix.m23 ); // or mat.GetColumn(3);
yourTransform.rotation = yourMatrix.rotation;
yourTransform.localScale = yourMatrix.lossyScale;

This has some flipping issues with negative scales, fix is:

https://pixel.engineer/posts/matrix-decomposition-unity/
https://gist.github.com/atteneder/b6675c9a73860c00d795dcea7149e8d2

Vector3 tpos; Quaternion trot; Vector3 tscale;
yourMatrix.Decompose(out tpos, out trot, out tscale );
yourTransform.position = tpos;
yourTransform.rotation = trot;
yourTransform.localScale = tscale;
